packages/app-builder-lib/src/node-module-collector/nodeModulesCollector.ts

```diff
@@ -0,0 +1,120 @@
+    if (!node) {
+      node = {
+        name,
+        identName: name,
+        reference: key.match(/@?[^@]+@?(.+)?/)![1] || ``,
+        dependencies: new Set<HoisterTree>(),
+        peerNames: new Set<string>([]),
+      }
+      nodes.set(key, node)
+
+      for (const dep of (obj[key] || {}).dependencies || []) {
+        node.dependencies.add(this.transToHoisterTree(obj, dep, nodes))
+      }
+    }
+    return node
+  }
+
+  private resolvePath(filePath: string) {
+    try {
+      const stats = fs.lstatSync(filePath)
+      if (stats.isSymbolicLink()) {
+        return fs.realpathSync(filePath)
+      } else {
+        return filePath
+      }
+    } catch (error) {
+      log.error({ filePath, error }, "Error resolving path")
+      return filePath
+    }
+  }
+
+  public TransToDependencyGraph(tree: DependencyTree): DependencyGraph {
+    const result: DependencyGraph = { ".": {} }
+
+    const flatten = (node: DependencyTree, parentKey = ".") => {
+      const dependencies = node.dependencies || {}
+
+      for (const [key, value] of Object.entries(dependencies)) {
+        // Skip empty dependencies(like some optionalDependencies)
+        if (Object.keys(value).length === 0) {
+          continue
+        }
+        const version = value.version || ""
+        const newKey = `${key}@${version}`
+        this.dependencyPathMap.set(newKey, path.normalize(this.resolvePath(value.path)))
+        if (!result[parentKey]?.dependencies) {
+          result[parentKey] = { dependencies: [] }
+        }
+        result[parentKey].dependencies!.push(newKey)
+        flatten(value, newKey)
+      }
+    }
+
+    flatten(tree)
+    return result
+  }
+
+  abstract getDependenciesTree(): DependencyTree
+  abstract getPMCommand(): string
+
+  private _getNodeModules(dependencies: Set<HoisterResult>, result: NodeModuleInfo[]) {
+    if (dependencies.size === 0) return
+
+    for (const d of dependencies.values()) {
+      const reference = [...d.references][0]
+      const p = this.dependencyPathMap.get(`${d.name}@${reference}`)
+      const node = {
+        name: d.name,
+        version: reference,
+        dir: p,
+      } as NodeModuleInfo
+      result.push(node)
+      if (d.dependencies.size > 0) {
+        node["dependencies"] = []
+        this._getNodeModules(d.dependencies, node["dependencies"])
+      }
+    }
+    result.sort((a, b) => a.name.localeCompare(b.name))
+  }
+
+  private getTreeFromWorkspaces(tree: DependencyTree): DependencyTree {
+    if (tree.workspaces && tree.dependencies) {
+      for (const [key, value] of Object.entries(tree.dependencies)) {
+        if (this.rootDir.endsWith(path.normalize(key))) {
+          return value
+        }
+      }
+    }
+    return tree
+  }
+
+  public getNodeModules(): NodeModuleInfo[] {
+    const tree = this.getDependenciesTree()
+    const realTree = this.getTreeFromWorkspaces(tree)
+    const dependencyGraph = this.TransToDependencyGraph(realTree)
```

Comment:
Can we rename this function? Not sure what `Trans` stands for and it also isn't camelcase
Perhaps `this.convertToDependencyGraph(realTree)`?